src/Microsoft.Extensions.SecretManager.Tools/Program.cs

```diff
@@ -147,27 +143,80 @@ internal int RunInternal(params string[] args)
             var projectFile = finder.FindMsBuildProject(projectPath);
 
             Logger.LogDebug(Resources.Message_Project_File_Path, projectFile);
 
+            var targetFile = GetTargetFile();
+            var outputFile = Path.GetTempFileName();
+
             try
             {
-                var project = new MsBuildProjectContextBuilder()
-                    .UseMsBuild(_msBuildContext)
-                    .AsDesignTimeBuild()
-                    .WithBuildTargets(Array.Empty<string>())
-                    .WithProjectFile(projectFile)
-                    .WithTargetFramework("") // TFM doesn't matter
-                    .Build();
-
-                return project.GetUserSecretsId();
+                var commandResult = Command.CreateDotNet("msbuild",
+                    new[] {
+                        targetFile,
+                        "/nologo",
+                        "/t:FindProperty",
+                        "/p:Project=" + projectFile,
+                        "/p:OutputFile=" + outputFile
+                    })
+                    .CaptureStdErr()
+                    .CaptureStdOut()
+                    .Execute();
+
+                if (commandResult.ExitCode != 0)
+                {
+                    throw new GracefulException(Resources.FormatError_ProjectFailedToLoad(projectFile));
+                }
+
+                var id = File.ReadAllText(outputFile).Trim();
+                if (string.IsNullOrEmpty(id))
+                {
+                    throw new GracefulException(Resources.FormatError_ProjectMissingId(projectFile));
+                }
+
+                return id;
+            }
+            finally
+            {
+                if (File.Exists(outputFile))
+                {
+                    File.Delete(outputFile);
+                }
             }
-            catch (InvalidProjectFileException ex)
+        }
+
+        private string GetTargetFile()
+        {
+            var targetFileName = "FindProperty.targets";
+            var assemblyDir = Path.GetDirectoryName(GetType().GetTypeInfo().Assembly.Location);
+            var searchPaths = new[] {
+                AppContext.BaseDirectory,
+                assemblyDir, // next to assembly
+                Path.Combine(assemblyDir, "../../tools"), // inside the nupkg
+            };
+
+            var foundFile = searchPaths
+                .Select(dir => Path.Combine(dir, targetFileName))
+                .Where(File.Exists)
+                .FirstOrDefault();
+
+            if (foundFile != null)
             {
-                throw new GracefulException(Resources.FormatError_ProjectFailedToLoad(projectFile), ex);
+                return foundFile;
             }
+
+            // back up plan - if the targets file was not found any of the expected locations,
+            // create the file using the embedded version
+            var outputPath = Path.Combine(AppContext.BaseDirectory, targetFileName);
```

Comment:
Currently used in testing. This can probably be removed later, but it's necessary as there isn't (yet) a good way to get the .targets file into the test project once we delete the src folder.